include/merlin_hashtable.cuh

```diff
@@ -411,9 +412,19 @@ class HashTable {
       return;
     }
 
-    while (!reach_max_capacity_ &&
-           fast_load_factor(n) > options_.max_load_factor) {
-      reserve(capacity() * 2);
+    std::shared_lock<std::shared_timed_mutex> lock(table_mutex_,
+                                                   std::defer_lock);
+    if (!reach_max_capacity_) {
+      // Attempt to grow if load factor is too high.
```

Comment:
I recommend remove these inline comments, and keep the code clean. 'the best code is self-documenting." BTW, I think your code has clear logic :)